### ISSUE TITLE
Screensaver / Screen-blanking prevention doesn't work when using password protected screensaver

### ISSUE THREAD

**Issue by Sheado** [glfw/glfw#851]
Fullscreen GLFW window does not prevent screensaver or screen sleep if the screen saver is set to show the logon screen on wake. Verified on Windows 10, 8.1.

The recommended solution:
Call SetThreadExecutionState function with ES_CONTINUOUS|ES_SYSTEM_REQUIRED argument.

More details on the issue can be found in this discussion:
http://discourse.glfw.org/t/screen-saver-lock-while-in-fullscreen/740/5

**Comment by tombsar** [comment]
Just to clarify, do you mean that the screensaver kicks in despite the user pressing keys or moving the mouse?

**Comment by Sheado** [comment]
@tombsar The screensaver is kicking in during fullscreen mode when there is no user keyboard/mouse input. In fullscreen mode, the screensaver is supposed to be suppressed regardless of whether there is user input or not (e.g. playing a game with a controller or watching a video). 

The following solution worked for me:
```
void setKeepScreenOn(bool isKeepOn)
{
	if ( isKeepOn )
		SetThreadExecutionState(ES_CONTINUOUS | ES_DISPLAY_REQUIRED);
	else
		SetThreadExecutionState(ES_CONTINUOUS);
}
```

**Comment by elmindreda** [comment]
This should be fixed now with 355b46e6c595341c6061c7bb6fd6131eecdc5bee. Thank you for reporting this!
